pkg/accounting/mock/accounting.go

```diff
@@ -1,49 +1,136 @@
+		s.creditFunc = f
+	})
+}
 
+// WithDebitFunc sets the mock Debit function
+func WithDebitFunc(f func(peer swarm.Address, price uint64) error) Option {
+	return optionFunc(func(s *Service) {
+		s.debitFunc = f
+	})
+}
+
+// WithBalanceFunc sets the mock Balance function
+func WithBalanceFunc(f func(swarm.Address) (int64, error)) Option {
+	return optionFunc(func(s *Service) {
+		s.balanceFunc = f
+	})
+}
+
+// WithBalancesFunc sets the mock Balances function
+func WithBalancesFunc(f func() (map[string]int64, error)) Option {
+	return optionFunc(func(s *Service) {
+		s.balancesFunc = f
+	})
+}
+
+// NewAccounting creates the mock accounting implementation
+func NewAccounting(opts ...Option) accounting.Interface {
+	mock := new(Service)
+	mock.balances = make(map[string]int64)
+	for _, o := range opts {
+		o.apply(mock)
+	}
+	return mock
 }
 
-func (ma *MockAccounting) Credit(peer swarm.Address, price uint64) error {
-	ma.lock.Lock()
-	defer ma.lock.Unlock()
-	ma.balances[peer.String()] -= int64(price)
+// Reserve is the mock function wrapper that calls the set implementation
+func (s *Service) Reserve(peer swarm.Address, price uint64) error {
+	if s.reserveFunc != nil {
+		return s.reserveFunc(peer, price)
+	}
+	return nil
+}
+
+// Release is the mock function wrapper that calls the set implementation
+func (s *Service) Release(peer swarm.Address, price uint64) {
+	if s.releaseFunc != nil {
+		s.releaseFunc(peer, price)
+	}
+}
+
+// Credit is the mock function wrapper that calls the set implementation
+func (s *Service) Credit(peer swarm.Address, price uint64) error {
+	if s.creditFunc != nil {
+		return s.creditFunc(peer, price)
+	}
+	s.lock.Lock()
+	defer s.lock.Unlock()
+	s.balances[peer.String()] -= int64(price)
 	return nil
 }
 
-func (ma *MockAccounting) Debit(peer swarm.Address, price uint64) error {
-	ma.lock.Lock()
-	defer ma.lock.Unlock()
-	ma.balances[peer.String()] += int64(price)
+// Debit is the mock function wrapper that calls the set implementation
+func (s *Service) Debit(peer swarm.Address, price uint64) error {
+	if s.debitFunc != nil {
+		return s.debitFunc(peer, price)
+	}
+	s.lock.Lock()
+	defer s.lock.Unlock()
+	s.balances[peer.String()] += int64(price)
 	return nil
 }
 
-func (ma *MockAccounting) Balance(peer swarm.Address) (int64, error) {
-	ma.lock.Lock()
-	defer ma.lock.Unlock()
-	return ma.balances[peer.String()], nil
+// Balance is the mock function wrapper that calls the set implementation
+func (s *Service) Balance(peer swarm.Address) (int64, error) {
+	if s.balanceFunc != nil {
+		return s.balanceFunc(peer)
+	}
+	s.lock.Lock()
+	defer s.lock.Unlock()
+	return s.balances[peer.String()], nil
 }
 
-func NewAccounting() *MockAccounting {
-	return &MockAccounting{
-		balances: make(map[string]int64),
+// Balances is the mock function wrapper that calls the set implementation
+func (s *Service) Balances() (map[string]int64, error) {
+	if s.balancesFunc != nil {
+		return s.balancesFunc()
 	}
+	return nil, nil
```

Comment:
could you please explain why this does not return `s.balances` if `s.balancesFunc == nil`?